Sources/Snowplow/Network/DefaultNetworkConnection.swift

```diff
@@ -88,13 +88,20 @@ public class DefaultNetworkConnection: NSObject, NetworkConnection {
     public var serverAnonymisation = false
     private var dataOperationQueue = OperationQueue()
     private var builderFinished = false
+    /// Custom timeout for the requests
+    @objc
+    public var timeout: NSNumber
+    private let defaultTimeout: NSNumber = 60
+    private var urlSession: URLSession?
     
     @objc
     public init(urlString: String,
                 httpMethod: HttpMethodOptions = EmitterDefaults.httpMethod,
                 protocol: ProtocolOptions = EmitterDefaults.httpProtocol,
-                customPostPath: String? = nil) {
+                customPostPath: String? = nil,
+                timeout: NSNumber? = nil) {
```

Comment:
This property does not give any indication about the unit of the timeout. I think it'd be better to use `TimeInterval` which is defined in seconds and more appropriate for time durations. We can't make `TimeInterval` optional (to keep compatibility with Obj-C), but we can set the default value here in the constructor like this:

```suggestion
                timeout: TimeInterval = 60) {
```